e3sm_diags/driver/utils/regrid.py

```diff
@@ -243,7 +242,7 @@ def _apply_land_sea_mask(
     ds_new = ds.copy()
     ds_new = _drop_unused_ilev_axis(ds)
     output_grid = ds_new.regridder.grid
-    mask_var_key = _get_region_mask_var_key(region)
+    mask_var_key = _get_region_mask_var_key(ds_mask, region)
```

Comment:
I ran into error below for a special usecase when the region is `land_60S90N`.
```
raise ValueError(f"Only land and ocean regions are supported, not '{region}'.")
ValueError: Only land and ocean regions are supported, not 'land_60S90N'.
```
It looks like we should have a way to distinguish land/ocean mask only and the other type of special regions.